modules/civiremote_event/src/Form/RegisterForm.php

```diff
@@ -724,41 +724,60 @@ protected function buildRegisterForm(array $form, FormStateInterface $form_state
     // Apply dependencies via #ajax.
     if (!empty($dependencies)) {
       foreach ($dependencies as $field_name => $field_dependencies) {
-        // Register an Ajax callback for the onChange event on the field.
         $field_group_parents = $this->groupParents($field_name);
         $field_exists = NULL;
         $field_group = &NestedArray::getValue($form, $field_group_parents, $field_exists);
         if ($field_exists) {
-          $field_group[$field_name]['#civiremote_event_dependencies'] = $field_dependencies;
-          $field_group[$field_name]['#ajax'] = [
-            'callback' => '::dependencyAjaxCallback',
-            'event' => 'change',
-            'effect' => 'fade',
-          ];
-          $field_group[$field_name]['#limit_validation_errors'] = [
-            [$field_name]
-          ];
-          $field_group[$field_name]['#submit'] = [[$this, 'submitForm']];
-
-          // Process dependent fields.
-          foreach ($field_dependencies as $dependent_field_name => $dependency) {
-            // Wrap dependent fields with a wrapper element.
-            $dependent_group_parents = $this->groupParents($field_name);
-            $dependent_group = &NestedArray::getValue($form, $dependent_group_parents);
-            if (empty($dependent_group[$dependent_field_name]['#prefix'])) {
-              $dependent_group[$dependent_field_name]['#prefix'] = '';
+          if ('hide' === $dependency['command']) {
+            // Use the States API for hiding the field.
+            $field_group[$field_name]['#states'] = [
+              'visible' => [
+                [':input[name="' . $dependency['dependee_field'] . '"]' => ['value' => $dependency['dependee_value']]]
```

Comment:
Is it possible to use the dependency mechanism with checkboxes? Then the condition would need to use `checked` instead of `value`.

Reply:
Then I'd suggest to add a todo comment.